src/main/java/org/commcare/cases/instance/FixtureIndexSchema.java

```diff
@@ -0,0 +1,81 @@
+package org.commcare.cases.instance;
+
+import org.commcare.cases.model.StorageIndexedTreeElementModel;
+import org.javarosa.core.model.data.IAnswerData;
+import org.javarosa.core.model.instance.TreeElement;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.Vector;
+
+/**
+ * Tracks what attributes and elements are stored in indexed columns of an
+ * indexed fixture db table.
+ *
+ * @author Phillip Mates (pmates@dimagi.com)
+ */
+public class FixtureIndexSchema {
+    private final Set<String> indices;
+    public final String fixtureName;
+
+    public FixtureIndexSchema(TreeElement schemaTree) {
+        this.fixtureName = schemaTree.getAttributeValue(null, "id");
+
+        indices = buildIndices(schemaTree.getChildrenWithName("index"));
+    }
+
+    private static Set<String> buildIndices(Vector<TreeElement> indexElements) {
```

Comment:
Super nitpicking, but this could just be an instance method called `setIndices` instead of a static method that returns something